src/commands/contract/verify/mod.rs

```diff
@@ -0,0 +1,277 @@
+            ).red())
+        }
+
+        Ok(Self)
+    }
+}
+
+fn verify_contract(
+    contract_code_hash_from_repository: near_verify_rs::types::sha256_checksum::SHA256Checksum,
+    contract_code_hash: near_primitives::hash::CryptoHash,
+) {
+    if contract_code_hash_from_repository.hash == contract_code_hash.as_bytes() {
+        tracing::info!("\n{}", crate::common::indent_payload(&
+            format!(
+            "The hash code obtained from the contract account ID and the hash code calculated from the repository are the same.\nhash: {}",
+            contract_code_hash
+        )))
+    } else {
+        tracing::info!("\n{}", crate::common::indent_payload(&
+            format!(
+            "The hash code obtained from the contract account ID: {}\nThe hash code calculated from the repository:        {}",
+            contract_code_hash,
+            contract_code_hash_from_repository.to_base58_string()
+        )).red())
+    };
+}
+
+#[tracing::instrument(
+    name = "Getting the contract code hash from the repository ...",
+    skip_all
+)]
+fn get_contract_code_hash_from_repository(
+    account_id: &near_primitives::types::AccountId,
+    network_config: &crate::config::NetworkConfig,
+    block_reference: &near_primitives::types::BlockReference,
+) -> color_eyre::eyre::Result<near_verify_rs::types::sha256_checksum::SHA256Checksum> {
+    let contract_source_metadata = tokio::runtime::Runtime::new().unwrap().block_on(
+        super::inspect::get_contract_source_metadata(
+            &network_config.json_rpc_client(),
+            block_reference,
+            account_id,
+        ),
+    )?;
+    let (_tempdir, docker_build_out_wasm) =
+        get_docker_build_out_wasm_from_contract_source_metadata(contract_source_metadata)?;
+
+    near_verify_rs::logic::compute_hash(docker_build_out_wasm)
+}
+
+#[tracing::instrument(
+    name = "Getting the docker build out wasm from the contract source metadata ...",
+    skip_all
+)]
+fn get_docker_build_out_wasm_from_contract_source_metadata(
+    contract_source_metadata: ContractSourceMetadata,
+) -> color_eyre::eyre::Result<(tempfile::TempDir, camino::Utf8PathBuf)> {
+    let build_info = contract_source_metadata.build_info.as_ref().wrap_err("`contract_source_metadata` does not have a `build_info` field. This field is an addition to version **1.2.0** of **NEP-330**.")?;
+    let source_id = SourceId::from_url(&build_info.source_code_snapshot)?;
+
+    let tempdir = tempfile::tempdir()?;
+    let target_dir = tempdir.path().to_path_buf();
+
+    let SourceKind::Git(GitReference::Rev(rev)) = source_id.kind();
+
+    checkout_remote_repo(source_id.url().as_str(), &target_dir, rev)?;
+
+    let target_dir = camino::Utf8PathBuf::from_path_buf(target_dir)
+        .map_err(|err| color_eyre::eyre::eyre!("convert path buf {:?}", err))?;
+
+    contract_source_metadata.validate(None)?;
```

Comment:
This has to use trusted images whitelist. See examples in `near_verify_rs` repo (at least `sourcescan/cargo-near`).

Reply:
Empty whitelist can be under a `#[interactive_clap(long, skip_interactive_input)] no_image_whitelist: bool` flag (imo)

Reply:
Done